File: go/test/protocol/test_case_action_invoke_command.go
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
package protocol

import (
	"gopkg.in/yaml.v3"
)

type testCaseActionInvokeCommand struct {
	InvocationIndex int                `yaml:"invocation-index"`
	CommandName     *string            `yaml:"command-name"`
	TopicTokenMap   map[string]string  `yaml:"topic-token-map"`
	Timeout         *TestCaseDuration  `yaml:"timeout"`
	RequestValue    *string            `yaml:"request-value"`
	Metadata        *map[string]string `yaml:"metadata"`
}

type TestCaseActionInvokeCommand struct {
	testCaseActionInvokeCommand
}

func (invokeCommand *TestCaseActionInvokeCommand) UnmarshalYAML(
	node *yaml.Node,
) error {
	*invokeCommand = TestCaseActionInvokeCommand{}

	invokeCommand.CommandName = TestCaseDefaultInfo.Actions.InvokeCommand.GetCommandName()
	invokeCommand.RequestValue = TestCaseDefaultInfo.Actions.InvokeCommand.GetRequestValue()

	err := node.Decode(&invokeCommand.testCaseActionInvokeCommand)

	if invokeCommand.Timeout == nil {
		defaultTimeout := TestCaseDefaultInfo.Actions.InvokeCommand.GetTimeout()
		invokeCommand.Timeout = &defaultTimeout
	}

	return err
}

File: dotnet/test/Azure.Iot.Operations.Protocol.MetlTests/TestCaseActionInvokeCommand.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

namespace Azure.Iot.Operations.Protocol.MetlTests
{
    public class TestCaseActionInvokeCommand : TestCaseAction
    {
        public static string? DefaultCommandName;
        public static string? DefaultRequestValue;
        public static TestCaseDuration? DefaultTimeout;

        public int? InvocationIndex { get; set; }

        public string? CommandName { get; set; } = DefaultCommandName;

        public Dictionary<string, string>? TopicTokenMap { get; set; }

        public TestCaseDuration? Timeout { get; set; } = DefaultTimeout;

        public string? RequestValue { get; set; } = DefaultRequestValue;

        public Dictionary<string, string>? Metadata { get; set; }
    }
}
